### PR TITLE
[AzureMonitorOpenTelemetryDistro] Add support for logs

### PR DESCRIPTION
**Pull request by rajkumar-rangaraj**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/monitor/Azure.Monitor.OpenTelemetry/tests/Azure.Monitor.OpenTelemetry.Demo/Program.cs [hunk 1]
```diff
@@ -15,7 +16,11 @@
 });
 
 var app = builder.Build();
-app.MapGet("/", () => $"Hello World! OpenTelemetry Trace: {Activity.Current?.Id}");
+app.MapGet("/", () =>
+{
+    app.Logger.LogError("Hello World!");
```
Comment (vishweshbankwar) [inline sdk/monitor/Azure.Monitor.OpenTelemetry/tests/Azure.Monitor.OpenTelemetry.Demo/Program.cs:21]: nit: Information?

File: sdk/monitor/Azure.Monitor.OpenTelemetry/src/AzureMonitorOpenTelemetryImplementations.cs [hunk 2]
```diff
@@ -37,6 +38,26 @@ internal static IServiceCollection AddAzureMonitorOpenTelemetryWithAction(IServi
...
+                }
+
+                if (logExporterOptions.EnableLogs)
+                {
+                    logging.AddOpenTelemetry(builderOptions =>
+                    {
+                        builderOptions.IncludeFormattedMessage = true;
+                        builderOptions.ParseStateValues = true;
+                        builderOptions.IncludeScopes = true;
+                        builderOptions.AddAzureMonitorLogExporter(o => logExporterOptions.SetValueToExporterOptions(o));
```
Comment (cijothomas) [inline sdk/monitor/Azure.Monitor.OpenTelemetry/src/AzureMonitorOpenTelemetryImplementations.cs:56]: could you check if users will still be able to add a separate exporter like Console or OTLP?

Comment (rajkumar-rangaraj) [reply sdk/monitor/Azure.Monitor.OpenTelemetry/src/AzureMonitorOpenTelemetryImplementations.cs:56]: Adding below configuration sends data to both Azure Monitor and Console.
```
builder.Services.Configure<OpenTelemetryLoggerOptions>(options =>
{
    options.AddConsoleExporter();
});
```

File: sdk/monitor/Azure.Monitor.OpenTelemetry/src/AzureMonitorOpenTelemetryImplementations.cs [hunk 3]
```diff
@@ -37,6 +38,26 @@ internal static IServiceCollection AddAzureMonitorOpenTelemetryWithAction(IServi
...
+                    configureAzureMonitorOpenTelemetry(logExporterOptions);
+                }
+
+                if (logExporterOptions.EnableLogs)
+                {
+                    logging.AddOpenTelemetry(builderOptions =>
+                    {
+                        builderOptions.IncludeFormattedMessage = true;
+                        builderOptions.ParseStateValues = true;
+                        builderOptions.IncludeScopes = true;
```
Comment (cijothomas) [inline sdk/monitor/Azure.Monitor.OpenTelemetry/src/AzureMonitorOpenTelemetryImplementations.cs:55]:
```suggestion
                        builderOptions.IncludeScopes = false;
```

Scopes add a lot of extra information, a lot of them from asp.net core itself which are duplicate info.. I'd vote to keep this opt-in behavior.

File: sdk/monitor/Azure.Monitor.OpenTelemetry/src/AzureMonitorOpenTelemetryOptions.cs [hunk 4]
```diff
@@ -66,9 +66,15 @@ internal AzureMonitorOpenTelemetryOptions Clone(AzureMonitorOpenTelemetryOptions
         internal void SetValueToExporterOptions(IServiceProvider sp)
         {
             var exporterOptions = sp.GetRequiredService<IOptionsMonitor<AzureMonitorExporterOptions>>().Get("");
+            SetValueToExporterOptions(exporterOptions);
+        }
+
+        internal AzureMonitorExporterOptions SetValueToExporterOptions(AzureMonitorExporterOptions exporterOptions)
```
Comment (vishweshbankwar) [inline sdk/monitor/Azure.Monitor.OpenTelemetry/src/AzureMonitorOpenTelemetryOptions.cs:72]: `void`

Comment (rajkumar-rangaraj) [reply sdk/monitor/Azure.Monitor.OpenTelemetry/src/AzureMonitorOpenTelemetryOptions.cs:72]: This has to return `AzureMonitorExporterOptions`, it is used in 2 places. The usage in `sdk/monitor/Azure.Monitor.OpenTelemetry/src/AzureMonitorOpenTelemetryImplementations.cs` requires return as `AzureMonitorExporterOptions`

**Review by vishweshbankwar — APPROVED** [review]

**Pull request merged**